pkg/provider/azure_local_services.go

```diff
@@ -544,3 +543,84 @@ func (az *Cloud) cleanupLocalServiceBackendPool(
 	}
 	return lbs, nil
 }
+
+// checkAndApplyLocalServiceBackendPoolUpdates if the IPs in the backend pool are aligned
+// with the corresponding endpointslice, and update the backend pool if necessary.
+func (az *Cloud) checkAndApplyLocalServiceBackendPoolUpdates(lb network.LoadBalancer, service *v1.Service) error {
+	serviceName := getServiceName(service)
+	endpointsNodeNames, err := az.getLocalServiceEndpointsNodeNames(service)
+	if err != nil {
+		return err
+	}
+	var expectedIPs []string
+	for nodeName := range endpointsNodeNames {
+		ips := az.nodePrivateIPs[strings.ToLower(nodeName)]
+		for ip := range ips {
+			expectedIPs = append(expectedIPs, ip)
+		}
+	}
+	currentIPsInBackendPools := make(map[string][]string)
+	for _, bp := range *lb.BackendAddressPools {
+		bpName := pointer.StringDeref(bp.Name, "")
+		if localServiceOwnsBackendPool(serviceName, bpName) {
+			var currentIPs []string
+			for _, address := range *bp.LoadBalancerBackendAddresses {
+				currentIPs = append(currentIPs, *address.IPAddress)
+			}
+			currentIPsInBackendPools[bpName] = currentIPs
+		}
+	}
+	az.applyIPChangesAmongLocalServiceBackendPoolsByIPFamily(*lb.Name, serviceName, currentIPsInBackendPools, expectedIPs)
+
+	return nil
+}
+
+// applyIPChangesAmongLocalServiceBackendPoolsByIPFamily reconciles IPs by IP family
+// amone the backend pools of a local service.
+func (az *Cloud) applyIPChangesAmongLocalServiceBackendPoolsByIPFamily(
+	lbName, serviceName string,
+	currentIPsInBackendPools map[string][]string,
+	expectedIPs []string,
+) {
+	currentIPsInBackendPoolsIPv4 := make(map[string][]string)
+	currentIPsInBackendPoolsIPv6 := make(map[string][]string)
+	for bpName, ips := range currentIPsInBackendPools {
+		if managedResourceHasIPv6Suffix(bpName) {
+			currentIPsInBackendPoolsIPv6[bpName] = ips
+		} else {
+			currentIPsInBackendPoolsIPv4[bpName] = ips
+		}
+	}
+
+	var ipv4, ipv6 []string
+	for _, ip := range expectedIPs {
+		if utilnet.IsIPv6String(ip) {
+			ipv6 = append(ipv6, ip)
+		} else {
+			ipv4 = append(ipv4, ip)
+		}
+	}
+	az.reconcileIPsInLocalServiceBackendPoolsAsync(lbName, serviceName, currentIPsInBackendPoolsIPv6, ipv6)
```

Comment:
same here, how is the error propagated back to the caller?

Reply:
It will be in the service events.